django/chat/static/chat/js/scripts.js

```diff
@@ -75,41 +75,43 @@ function render_markdown(element) {
 }
 
 // Chat window UI
-let preventAutoScrolling = false;
-let ignoreNextScrollEvent = true;
+let autoscroll = true;
+const scrollBtn = document.querySelector("#scroll-btn");
+
+document.querySelector("#chat-container").addEventListener("scroll", function () {
+  const threshold = 10; // pixels from the bottom considered "at the bottom"
+  if ((this.scrollHeight - this.scrollTop - this.clientHeight) > threshold) {
+    autoscroll = false;
+    scrollBtn.classList.add("show");
+  } else {
+    autoscroll = true;
+    scrollBtn.classList.remove("show");
+  }
+});
 
 const copyCodeButtonHTML = `<button type="button" onclick="copyCode(this)"
 class="btn btn-link m-0 p-0 text-muted copy-message-button copy-button"
 title="Copy"><i class="bi bi-copy"></i><i class="bi bi-check-lg"></i></button>`;
 
 function scrollToBottom(smooth = true, force = false) {
   resizePromptContainer();
-  if (preventAutoScrolling && !force) {
+  if (!autoscroll && !force) {
     return;
   }
-
-  ignoreNextScrollEvent = true;
-
   let messagesContainer = document.querySelector("#chat-container");
-  let hashContainer = null;
-  let hashRect = null;
-  let containerRect = messagesContainer.getBoundingClientRect();
-
-  if (window.location.hash) {
-    hashContainer = document.querySelector(window.location.hash);
-    hashRect = hashContainer.getBoundingClientRect();
+  let destination = messagesContainer.scrollHeight;
+  // If there is currently a response streaming, disable smooth
+  if (document.querySelector(".chat-streaming-response")) {
+    smooth = false;
```

Comment:
Smooth scrolling while a response was happening meant that it would never reach the bottom. Simple fix is to just jump straight to the bottom in this case.